SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -533,3 +541,88 @@ void ASpatialDebugger::SpatialToggleDebugger()
 		}
 	}
 }
+
+void ASpatialDebugger::EditorSpatialToggleDebugger(bool bEnabled)
+{
+	bShowWorkerRegions = bEnabled;
+	EditorRefreshWorkerRegions();
+}
+
+void ASpatialDebugger::EditorRefreshWorkerRegions()
+{
+	DestroyWorkerRegions();
+
+	if (bShowWorkerRegions && EditorAllowWorkerBoundaries())
+	{
+		EditorInitialiseWorkerRegions();
+		CreateWorkerRegions();
+	}
+#if WITH_EDITOR
+	if (GEditor != nullptr && GEditor->GetActiveViewport() != nullptr)
+	{
+		// Redraw editor window to show changes
+		GEditor->GetActiveViewport()->Invalidate();
+	}
+#endif
+}
+
+bool ASpatialDebugger::EditorAllowWorkerBoundaries() const
+{
+#if WITH_EDITOR
+	// Check if multi worker is enabled.
+	UWorld* World = GetWorld();
+	check(World != nullptr);
+
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings());
+	const bool bIsMultiWorkerEnabled = USpatialStatics::IsSpatialMultiWorkerEnabled(World);
+	const bool bIsSpatialNetworkingEnabled = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();
+
+	return bIsMultiWorkerEnabled && bIsSpatialNetworkingEnabled;
+#else
+	return false;
+#endif
+}
+
+void ASpatialDebugger::EditorInitialiseWorkerRegions()
+{
+	WorkerRegions.Empty();
+
+	const UWorld* World = GetWorld();
+	check(World != nullptr);
+
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings());
+	check(WorldSettings != nullptr);
+
+	const TSubclassOf<UAbstractSpatialMultiWorkerSettings> MultiWorkerSettingsClass = *WorldSettings->MultiWorkerSettingsClass;
+
+	const UAbstractSpatialMultiWorkerSettings* MultiWorkerSettings =
+		NewObject<UAbstractSpatialMultiWorkerSettings>(this, *MultiWorkerSettingsClass);
+
+	ULayeredLBStrategy* LoadBalanceStrategy = NewObject<ULayeredLBStrategy>(this);
```

Comment:
We're not explicitly destroying these objects at any point - doesn't that mean we're duplicated them if this function is called multiple times?

Reply:
It is my understanding that UObjects created with NewObject are automatically garbage collected when the object is no longer referenced. In this case, as they are local variables they should go out of scope at the end of the function.

Reply:
Added the following at the end of the function to ensure these objects are destroyed:
```
	MultiWorkerSettings->ConditionalBeginDestroy();
	MultiWorkerSettings = nullptr;
	LoadBalanceStrategy->ConditionalBeginDestroy();
	LoadBalanceStrategy = nullptr;
```

Reply:
Changed to create them with the default transient package so that the explicit destroy is not required.